tests/test_run.py

```diff
@@ -42,3 +44,40 @@ def test_run_script_from_internet(pipx_temp_env, capsys):
             "6bdfbb6e9c1132b1c38fdd2f195d4a24c540c324/pipx-demo.py",
         ]
     )
+
+
+def test_appargs_doubledash(pipx_temp_env, capsys, monkeypatch):
+    parser = pipx.main.get_command_parser()
+
+    input_argv = [
+        ["pipx", "run", "pycowsay", "--", "hello"],
+        ["pipx", "run", "pycowsay", "--", "--", "hello"],
```

Comment:
An intuitive interface to `pipx`, IMO, with be to support the `--` argument like nox does. pipx would forward anything after the first `--` to the app. So `pipx nox pycowsay -- hello` would say "hello", not "-- hello".

`pipx run nox -- -s lint` would run `nox -s lint`

and `pipx run nox -- -s lint -- nox_posargs` would run `nox -s lint -- nox_posargs`. It looks like currently the way it's implemented in this PR, it  would pass `nox -- -s lint -- nox_posargs`.

Reply:
I think that's how it works now.  But I am confused, because before I was trying to use a `pipx run` with nox in the past, and it seemed like pipx was removing the `--` wherever it was.  But now while writing the tests, it appears that pipx only removes the first `--` if it is not inside the appargs.  i.e. it may only remove the first argument after the app name if it is `--`.

So somehow it seems to behave differently now than when I was testing it a week or so ago.  I don't know if that's my mistake or code difference.

The other thing is that to my mind things get confusing if pipx or its argparse is doctoring anything after `pipx run <appname>`  In my mind I really expect that all to go verbatim to the app, as if the command line was `<appname> [...]` for `pipx run <appname [...]`.

If we told the user to always put a `--` before `<appname>` then it would basically work fine without any code modification.  That was one way I was thinking about addressing this, with just specifying the help text.